src/routes/categories/containers/CategoriesPage.js

```diff
@@ -73,35 +102,55 @@ class CategoriesPage extends React.Component {
   }
 
   render () {
-    const url = this.getParentPath()
-    const hierarchy = new Hierarchy(this.props.path)
+    const category = CategoryModel.getCategoryByPath(this.props.categories, this.props.path)
+    if (!category) {
+      // todo show an error that the page has not been found, redirect to the notFound route
+    }
+    const children = this.props.categories.filter(_category => category.children.includes(_category.id))
 
-    // Pass data to hierarchy
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      return <Error error={error} />
+    let Content
+    if (category.children.length === 0) {
+      // last level, our category is a simple page
+      Content = <Page page={category} />
+    } else if (category.id === 0) {
+      // first level, we want to display a table with all first order categories
+      Content = <CategoryTiles categories={children}
+                               title={category.title}
+                               locations={this.props.locations} />
+    } else {
+      // some level between, we want to display a list
+      Content = <TitledContentList categories={children}
+                                   parentCategory={category} />
     }
 
     return <div>
       <Breadcrumb
-        hierarchy={hierarchy}
-        language={this.props.language}
-        location={this.props.location}
-      />
-      <Content url={url} hierarchy={hierarchy} />
+        categories={this.props.categories}
+        category={category}
+        locations={this.props.locations} />
+      {Content}
       <PdfButton href={this.getPdfFetchPath()} />
     </div>
   }
 }
 
+const mapDispatchToProps = (dispatch) => ({
+  dispatchLanguageChangeUrls: (urls, languages, availableLanguages) => dispatch(
+    setLanguageChangeUrls(urls, languages, availableLanguages)
+  ),
+  dispatchRedirectToUrl: (url) => dispatch(replace(url))
+})
+
 const mapStateToProps = (state) => ({
   language: state.router.params.language,
   location: state.router.params.location,
-  path: state.router.params['_'] // _ contains all the values from *
+  path: state.router.params['_'], // _ contains all the values from *
```

Comment:
You probably want to use `path: state.router.pathname` now which contains not only the values from *, but the whole path.

Reply:
👍 this is similar to EventsPage now

Reply:
i should  have read this before I removed the baseUrl from CategoryModel again....